packages/apollo-link-retry/src/__tests__/retryLink.ts

```diff
@@ -145,6 +145,7 @@ describe('RetryLink', () => {
     const attemptStub = jest.fn();
     attemptStub.mockReturnValueOnce(true);
     attemptStub.mockReturnValueOnce(true);
+    attemptStub.mockReturnValueOnce(Promise.resolve(true));
```

Comment:
Nit here, but let's break this out into a separate test, so it's easier to focus on if someone breaks logic related to promise support in the future